miio/integrations/vacuum/dreame/dreamevacuum_miot.py

```diff
@@ -190,12 +256,115 @@ def volume(self) -> str:
         return self.data["voice_package"]
 
+    @property
+    def timezone(self) -> str:
+        return self.data["timezone"]
 
-class DreameVacuumMiot(MiotDevice):
-    """Interface for Vacuum 1C STYTJ01ZHM (dreame.vacuum.mc1808)"""
+    @property
+    def cleaning_time(self) -> str:
+        return self.data["cleaning_time"]
+
+    @property
+    def cleaning_area(self) -> str:
+        return self.data["cleaning_area"]
+
+    @property
+    def first_clean_time(self) -> str:
+        return self.data["first_clean_time"]
+
+    @property
+    def total_clean_time(self) -> str:
+        return self.data["total_clean_time"]
+
+    @property
+    def total_clean_times(self) -> str:
+        return self.data["total_clean_times"]
+
+    @property
+    def total_clean_area(self) -> str:
+        return self.data["total_clean_area"]
+
+    @property
+    def cleaning_mode(self):
+        cleaning_mode = self.data["cleaning_mode"]
+        if self.model == DREAME_1C:
+            try:
+                return CleaningModeDreame1C(cleaning_mode)
+            except ValueError:
+                _LOGGER.error(f"Unknown CleaningMode ({cleaning_mode})")
+                return CleaningModeDreame1C.Unknown
+        elif self.model in [DREAME_D9, DREAME_F9]:
+            try:
+                return CleaningModeDreameF9(cleaning_mode)
+            except ValueError:
+                _LOGGER.error(f"Unknown CleaningMode ({cleaning_mode})")
+                return CleaningModeDreameF9.Unknown
 
-    mapping = _MAPPING
-    _supported_models = ["dreame.vacuum.mc1808"]
+    @property
+    def life_sieve(self) -> Optional[str]:
+        try:
+            return self.data["life_sieve"]
+        except KeyError:
+            return None
+
+    @property
+    def life_brush_side(self) -> Optional[str]:
+        try:
+            return self.data["life_brush_side"]
+        except KeyError:
+            return None
+
+    @property
+    def life_brush_main(self) -> Optional[str]:
+        try:
+            return self.data["life_brush_main"]
+        except KeyError:
+            return None
+
+    @property
+    def water_flow(self) -> Optional[WaterFlow]:
+        try:
+            water_flow = self.data["water_flow"]
+        except KeyError:
+            return None
+        try:
+            return WaterFlow(water_flow)
+        except ValueError:
+            _LOGGER.error("Unknown WaterFlow (%s)", self.data["water_flow"])
+            return WaterFlow.Unknown
+
+    @property
+    def is_water_box_carriage_attached(self) -> Optional[bool]:
+        """Return True if water box carriage (mop) is installed, None if sensor not
+        present."""
+        if "water_box_carriage_status" in self.data:
+            return self.data["water_box_carriage_status"] == 1
+        return None
+
+
+class DreameVacuumBase(MiotDevice):
+    _supported_models = [
+        DREAME_1C,
+        DREAME_D9,
+        DREAME_F9,
+    ]
+
+    mapping = MIOT_MAPPING[DREAME_1C]
```

Comment:
```suggestion
    mapping = MIOT_MAPPING
```
It is enough to define a dict of mappings, the correct mapping should be chosen now since #1302 is merged. If this isn't working, let me know and let's find a solution.